src/main/java/ch/njol/skript/expressions/ExprFunction.java

```diff
@@ -0,0 +1,157 @@
+		"set {_function} to the function named \"myFunction\"",
+		"run {_function} with arguments 13 and true"
+})
+@Since("INSERT VERSION")
+@SuppressWarnings("rawtypes")
+public class ExprFunction extends SimpleExpression<DynamicFunctionReference> {
+
+	static {
+		Skript.registerExpression(ExprFunction.class, DynamicFunctionReference.class, ExpressionType.SIMPLE,
+				"[the|a] function [named] %string% [local:(in|from) %-script%]",
+				"[the] functions [named] %strings% [local:(in|from) %-script%]",
+				"[all] [the] functions (in|from) %script%"
+		);
+	}
+
+	private Expression<String> name;
+	private Expression<Script> script;
+	private int mode;
+	private boolean local;
+	private Script here;
+
+	@SuppressWarnings("null")
+	@Override
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed,
+                        ParseResult result) {
+		if (!this.getParser().hasExperiment(Feature.SCRIPT_REFLECTION))
+			return false;
+		this.mode = matchedPattern;
+		this.local = result.hasTag("local") || mode == 2;
+		switch (mode) {
+			case 0, 1 -> {
+				//noinspection unchecked
+				this.name = (Expression<String>) expressions[0];
+				if (local)
+					//noinspection unchecked
+					this.script = (Expression<Script>) expressions[1];
+			}
+			case 2 ->
+				//noinspection unchecked
+				this.script = (Expression<Script>) expressions[0];
+		}
+		this.here = this.getParser().getCurrentScript();
+		return true;
+	}
+
+	@Override
+	protected DynamicFunctionReference<?>[] get(Event event) {
+		@Nullable Script script;
+		if (local)
+			script = this.script.getSingle(event);
+		else
+			script = here;
+		return switch (mode) {
+			case 0:
+				@Nullable String name = this.name.getSingle(event);
+				if (name == null)
+					yield CollectionUtils.array();
+				@Nullable DynamicFunctionReference reference = this.resolveFunction(name, script);
+				if (reference == null)
+					yield CollectionUtils.array();
+				yield CollectionUtils.array(reference);
+			case 1:
+				yield this.name.stream(event).map(string -> this.resolveFunction(string, script))
+						.filter(Objects::nonNull)
+						.toArray(DynamicFunctionReference[]::new);
+			case 2:
+				if (script == null)
+					yield CollectionUtils.array();
+				@Nullable Namespace namespace = Functions.getScriptNamespace(script.getConfig().getFileName());
+				if (namespace == null)
+					yield CollectionUtils.array();
+				yield namespace.getFunctions().stream()
+						.map(DynamicFunctionReference::new)
+						.toArray(DynamicFunctionReference[]::new);
+			default:
+				throw new IllegalStateException("Unexpected value: " + mode);
+		};
+	}
+
+	private @Nullable DynamicFunctionReference resolveFunction(String name, @Nullable Script script) {
+		// Function reference string-ifying appends a () and potentially its source,
+		// e.g. `myFunction() from MyScript.sk` and we should turn that into a valid function.
+		if (script == null && !local && name.contains(") from ")) {
```

Comment:
No, this is specifically for de-stringifying stuff (so that you could "parse" a function reference from what it spits out in `"%{func}%"`) because I thought that'd be a big QoL thing for a lot of users.
Ideally you resolve it with a Script reference, but it's possible that it's a local function like `blah() from myscript.sk` in which case you can't just retrieve it with its own name.